### PR TITLE
tsify router

### LINKED ISSUE #2001: TypeScript-ify: router

### PR DESCRIPTION
**Pull request by jeffbcross**
Fixes #2001

### CODE REVIEW

**Comment by jeffbcross** [comment]
@btford Travis is green!

File: modules/angular2/src/mock/location_mock.ts [hunk 1]
```diff
@@ -22,29 +22,17 @@ export class SpyLocation extends SpyObject {
...
-    ObservableWrapper.callNext(this._subject, {
-      'url': pathname
-    });
-  }
+  simulateUrlPop(pathname: string) { ObservableWrapper.callNext(this._subject, {'url': pathname}); }
 
-  normalizeAbsolutely(url) {
-    return this._baseHref + url;
-  }
+  normalizeAbsolutely(url) { return this._baseHref + url; }
```
Comment (btford) [inline modules/angular2/src/mock/location_mock.ts]: Any reason for these formatting changes?

Comment (jeffbcross) [inline modules/angular2/src/mock/location_mock.ts]: Clang did it

Comment (jeffbcross) [inline modules/angular2/src/mock/location_mock.ts]: Clang must have different ideas about .ts vs .js files

**Comment by btford** [comment]
LGTM – just curious about all the extraneous formating changes. Are these to appease some new compiler/formatter? Does the TypeScript compiler have opinions about how the code is formatted?

**Comment by jeffbcross** [comment]
@btford I only made formatting changes that clang told me to do. I think it may format slightly differently for .ts than .js

**Comment by jeffbcross** [comment]
Merged at https://github.com/angular/angular/commit/ba07f39347e3f3609f015f32083f90f5cca5ec77

**Comment by btford** [comment]
so proud :')

**Pull request closed without merging**